### ISSUE TITLE
Update schemaLocation to current version during the distribution upgrade

### ISSUE THREAD

**Issue by Holmistr** [hazelcast/hazelcast#21267]
During the startup of the upgraded member (I used `hz start` with latest `5.2-SNAPSHOT`, commit `d084672`), there's a warning:
```
[ WARN] [main] [c.h.c.AbstractXmlConfigHelper]: Name of the hazelcast schema location[http://www.hazelcast.com/schema/config/hazelcast-config-5.1.xsd] is incorrect, using default
```
We should minimize the amount of unnecessary warnings to not scare the users.

As discussed in https://hazelcast.slack.com/archives/C02JGKJD9NY/p1650439739820989 , the solution is to add an additional step updating `schemaLocation` in `config/*.xml` files to the newly installed version and let the XML validator do its job to make sure that XML structure is still valid against the current schema.

In other words, we need to modify our scripting for deb/rpm/brew packages to update `schemaLocation` in existing config files
